sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,343 @@
+                "sklearn": sklearn_RidgeRegression.fit,
+            },
+            X,
+            y,
+            sample_weight,
+        )
+        return self
+
+    @wrap_output_data
+    def predict(self, X):
+        if not hasattr(self, "coef_"):
+            msg = (
+                "This %(name)s instance is not fitted yet. Call 'fit' with "
+                "appropriate arguments before using this estimator."
+            )
+            raise NotFittedError(msg % {"name": self.__class__.__name__})
+
+        return dispatch(
+            self,
+            "predict",
+            {
+                "onedal": self.__class__._onedal_predict,
+                "sklearn": sklearn_RidgeRegression.predict,
+            },
+            X,
+        )
+
+    def _test_type_and_finiteness(self, X_in):
+        X = X_in if isinstance(X_in, np.ndarray) else np.asarray(X_in)
+
+        dtype = X.dtype
+        if "complex" in str(type(dtype)):
+            return False
+
+        try:
+            _assert_all_finite(X)
+        except BaseException:
+            return False
+        return True
+
+    def _onedal_fit_supported(self, device, method_name, *data):
+        assert method_name == "fit"
+        assert len(data) == 3
+        X, y, sample_weight = data
+
+        class_name = self.__class__.__name__
+        patching_status = PatchingConditionsChain(
+            f"sklearn.linear_model.{class_name}.fit"
+        )
+
+        normalize_is_set = (
+            hasattr(self, "normalize")
+            and self.normalize
+            and self.normalize != "deprecated"
+        )
+        positive_is_set = hasattr(self, "positive") and self.positive
+
+        n_samples = _num_samples(X)
+        n_features = _num_features(X, fallback_1d=True)
+
+        # Check if equations are well defined
+        is_underdetermined = n_samples < (n_features + int(self.fit_intercept))
+
+        dal_ready = patching_status.and_conditions(
+            [
+                (sample_weight is None, "Sample weight is not supported."),
+                (
+                    not issparse(X) and not issparse(y),
+                    "Sparse input is not supported.",
+                ),
+                (not normalize_is_set, "Normalization is not supported."),
+                (
+                    not positive_is_set,
+                    "Forced positive coefficients are not supported.",
+                ),
+                (
+                    not is_underdetermined,
+                    "The shape of X (fitting) does not satisfy oneDAL requirements:"
+                    "Number of features + 1 >= number of samples.",
+                ),
+            ]
+        )
+
+        if device == "gpu":
+            alpha_is_scalar = isinstance(self.alpha, (int, float))
+            dal_ready = patching_status.and_condition(
+                alpha_is_scalar,
+                "Non-scalar alpha is not supported for GPU.",
+            )
```

Comment:
Instead of passing the device (which is the purpose of _onedal_gpu_supported/_onedal_cpu_supported in the first place), start and pass the patchingconditionschain instead and move this check into _onedal_gpu_supported after the chain is initialized. You can remove device from the args, and replace it with the chain.  This will make things a little easier to follow.